ross/utils.py

```diff
@@ -0,0 +1,237 @@
+            if isinstance(row[i], str):
+                if not header_found:
+                    if row[i].lower() == header_key_word:
+                        header_index = index
+                        header_found = True
+                if 'inches' in row[i].lower() or 'lbm' in row[i].lower():
+                    convert_to_metric = True
+                if header_found and convert_to_metric:
+                    break
+        if header_found and convert_to_metric:
+            break
+    if not header_found:
+        sys.exit("Could not find the header. Make sure the table has a header "
+                 "containing the names of the columns. In the case of a " + element + ", "
+                 "there should be a column named " + header_key_word + ".")
+
+    # Get specific data from the file
+    new_materials = {}
+    if element == "shaft" and sheet_type == "Model":
+        material_header_index = -1
+        material_header_found = False
+        material_header_key_word = "matno"
+        for index, row in df.iterrows():
+            for i in range(0, row.size):
+                if isinstance(row[i], str):
+                    if row[i].lower() == material_header_key_word:
+                        material_header_index = index
+                        material_header_found = True
+                        break
+            if material_header_found:
+                break
+        if not material_header_found:
+            sys.exit("Could not find the header for the materials. Make sure the table has a header "
+                     "with the parameters for the materials that will be used. There should be a column "
+                     "named " + material_header_key_word + ".")
+        df_material = pd.read_excel(file, header=material_header_index, sheet_name=sheet_name)
+        material_name = []
+        material_rho = []
+        material_e = []
+        material_g_s = []
+        for index, row in df_material.iterrows():
+            if not pd.isna(row["matno"]):
+                material_name.append(int(row["matno"]))
+                material_rho.append(row["rhoa"])
+                material_e.append(row["ea"])
+                material_g_s.append(row["ga"])
+            else:
+                break
+        if convert_to_metric:
+            for i in range(0, len(material_name)):
+                material_rho[i] = material_rho[i] * 27679.904
+                material_e[i] = material_e[i] * 6894.757
+                material_g_s[i] = material_g_s[i] * 6894.757
+        for i in range(0, len(material_name)):
+            new_material = Material(
+                name="shaft_mat_" + str(material_name[i]),
+                rho=material_rho[i],
+                E=material_e[i],
+                G_s=material_g_s[i],
+            )
+            new_materials["shaft_mat_" + str(material_name[i])] = new_material
+
+    if not is_csv:
+        df = pd.read_excel(file, header=header_index, sheet_name=sheet_name)
+    else:
```

Comment:
I think that with this method we are not going to need to import from .csv files, so we could remove that part of the code.